evalml/utils/gen_utils.py

```diff
@@ -300,18 +311,26 @@ def _convert_woodwork_types_wrapper(pd_data):
         pd_data (pd.Series, pd.DataFrame, pd.ExtensionArray): Pandas data structure
 
     Returns:
-        New pandas data structure (pd.DataFrame or pd.Series) with original data and dtypes that can be handled by numpy
+        Modified pandas data structure (pd.DataFrame or pd.Series) with original data and dtypes that can be handled by numpy
     """
     nullable_to_numpy_mapping = {pd.Int64Dtype: 'int64',
                                  pd.BooleanDtype: 'bool',
                                  pd.StringDtype: 'object'}
-
+    nullable_to_numpy_mapping_nan = {pd.Int64Dtype: 'float64',
+                                     pd.BooleanDtype: 'object',
```

Comment:
Got it, so this is what you mentioned in the PR description [RE this thread in pandas-dev](https://github.com/pandas-dev/pandas/issues/30038)?

Reply:
Yup yup! In the case where there is a pd.NA value that needs to be handled, we need to convert to potentially different types (since int64 and bool can't handle).